molo/core/models.py

```diff
@@ -12,9 +14,31 @@
 from wagtail.wagtailimages.blocks import ImageChooserBlock
 
 from molo.core.blocks import MarkDownBlock
-
-
-class HomePage(Page):
+from molo.core import constants
+
+
+class CommentedPageMixin(object):
+    def get_effective_commenting_settings(self):
+        # return commenting settings for the homepage
+        if self.commenting_state:
+            return {
+                'state': self.commenting_state,
+                'open_time': self.commenting_open_time,
+                'close_time': self.commenting_close_time
+            }
+        # use the commenting settings for the parent page
+        parent_page = Page.objects.all().ancestor_of(self).last().specific
+        if (parent_page):
```

Comment:
Also, I'm not sure this check will ever be valid - if the page has no parent, `.last()` will return `None` and `.specific` will fail. So you'll need to call `.specific` after the check passes